File: app/src/main/java/com/communityx/adapters/DonatedCreatorAdapter.java
package com.communityx.adapters;

import android.app.Activity;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import butterknife.ButterKnife;
import com.communityx.R;

import java.util.ArrayList;

public class DonatedCreatorAdapter extends RecyclerView.Adapter<DonatedCreatorAdapter.EventViewHolder> {
    private ArrayList<String> mDonatedList;
    private Activity mActivity;
    private LayoutInflater mLayoutInflator;

    public DonatedCreatorAdapter(ArrayList<String> donatedList, Activity activity) {
        this.mDonatedList = donatedList;
        this.mActivity = activity;
        this.mLayoutInflator = LayoutInflater.from(activity);
    }

    @NonNull
    @Override
    public EventViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        View view = mLayoutInflator.inflate(R.layout.item_donated, viewGroup, false);
        return new DonatedCreatorAdapter.EventViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull EventViewHolder eventViewHolder, int i) {

    }

    @Override
    public int getItemCount() {
        return 10;
    }

    public class EventViewHolder extends RecyclerView.ViewHolder {

        public EventViewHolder(@NonNull View itemView) {
            super(itemView);
            ButterKnife.bind(this, itemView);
        }
    }
}

File: app/src/main/java/com/communityx/adapters/DonatedCreatorAdapter.kt
package com.communityx.adapters

import android.app.Activity
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import butterknife.ButterKnife
import com.communityx.R

import java.util.ArrayList

class DonatedCreatorAdapter(private val mDonatedList: ArrayList<String>, private val mActivity: Activity) :
    RecyclerView.Adapter<DonatedCreatorAdapter.EventViewHolder>() {

    private val mLayoutInflator: LayoutInflater = LayoutInflater.from(mActivity)

    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): EventViewHolder {
        val view = mLayoutInflator.inflate(R.layout.item_donated, viewGroup, false)
        return EventViewHolder(view)
    }

    override fun onBindViewHolder(eventViewHolder: EventViewHolder, i: Int) {

    }

    override fun getItemCount(): Int {
        return 10
    }

     class EventViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        init {
            ButterKnife.bind(this, itemView)
        }
    }
}
